packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,498 @@
+};
+
+/**
+ * Decrypt an encrypted credential bundle.
+ *
+ * @param {string} credentialBundle - The encrypted credential bundle.
+ * @param {string} embeddedKey - The private key for decryption.
+ * @returns {Promise<Uint8Array | null>} - The decrypted data or null if decryption fails.
+ */
+export const decryptBundle = async (
+  credentialBundle: string,
+  embeddedKey: string
+): Promise<Uint8Array | null> => {
+  try {
+    let bundleBytes: Uint8Array;
+
+    // Decode using either Base58Check or Base64URL
+    if (
+      credentialBundle.indexOf("-") === -1 &&
+      credentialBundle.indexOf("_") === -1 &&
+      credentialBundle.indexOf("O") === -1 &&
+      credentialBundle.indexOf("I") === -1 &&
+      credentialBundle.indexOf("l") === -1 &&
+      credentialBundle.indexOf("0") === -1
+    ) {
+      bundleBytes = bs58check.decode(credentialBundle);
+    } else {
+      bundleBytes = base64urlDecode(credentialBundle);
+    }
+
+    if (bundleBytes.byteLength <= 33) {
+      throw new Error(
+        `Bundle size ${bundleBytes.byteLength} is too low. Expecting a compressed public key (33 bytes) and an encrypted credential.`
+      );
+    }
+
+    const compressedEncappedKeyBuf = bundleBytes.slice(0, 33);
+    const ciphertextBuf = bundleBytes.slice(33);
+    const encappedKeyBuf = uncompressRawPublicKey(compressedEncappedKeyBuf);
+    const decryptedData = await hpkeDecrypt({
+      ciphertextBuf,
+      encappedKeyBuf,
+      receiverPriv: embeddedKey,
+    });
+
+    return decryptedData;
+  } catch (error) {
+    console.error("Error injecting bundle:", error);
+    return null;
+  }
+};
+
+/**
+ * Encode a Uint8Array to a base64url-encoded string.
+ *
+ * @param {Uint8Array} data - The data to encode.
+ * @returns {string} - The base64url-encoded string.
+ */
+export const base64urlEncode = (data: Uint8Array): string => {
+  let binary = "";
+  data.forEach((byte) => (binary += String.fromCharCode(byte)));
+  const base64String = btoa(binary);
+  return base64String
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_")
+    .replace(/=+$/, "");
+};
+
+/**
+ * Decode a base64url-encoded string to a Uint8Array.
+ *
+ * @param {string} base64url - The base64url-encoded string to decode.
+ * @returns {Uint8Array} - The decoded data.
+ */
+export const base64urlDecode = (base64url: string): Uint8Array => {
+  const base64 = base64url.replace(/-/g, "+").replace(/_/g, "/");
+  const padding =
+    base64.length % 4 === 0 ? "" : "===".slice(0, 4 - (base64.length % 4));
+  const binaryString = atob(base64 + padding);
+  const len = binaryString.length;
+  const bytes = new Uint8Array(len);
+  for (let i = 0; i < len; i++) {
+    bytes[i] = binaryString.charCodeAt(i);
+  }
+  return bytes;
+};
+
+/**
+ * Generate a target key pair using P-256.
+ *
+ * @returns {Promise<KeyPair>} - The generated key pair.
+ */
+export const generateTargetKey = async (): Promise<KeyPair> => {
```

Comment:
Tiny nit: let's add a comment either to this method or `generateP256KeyPair` that will further differentiate the two. We can also consider changing the return types of the two methods based on difference in use cases

Reply:
Good call!